src/main/java/com/dongtronic/diabot/commands/NightscoutCommand.kt

```diff
@@ -108,11 +107,19 @@ class NightscoutCommand(category: Command.Category) : DiabotCommand() {
 
     private fun getUnstoredData(event: CommandEvent) {
         val args = event.args.split("\\s+".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
-
-        val hostname = args[0]
-        val endpoint = "https://$hostname.herokuapp.com/api/v1/"
-
-        buildNightscoutResponse(endpoint, event)
+        if (event.event.message.mentionedUsers.size == 1) {
+            val user = event.event.message.mentionedMembers.get(0).user
+            val endpoint = getNightscoutHost(user) + "/api/v1/"
+            buildNightscoutResponse(endpoint, event)
+        } else if (event.event.message.mentionedUsers.size > 1) {
+            throw IllegalArgumentException("Too many mentioned users.")
+        } else if (event.event.message.mentionsEveryone()) {
+            throw IllegalArgumentException("Cannot handle mentioning everyone.")
+        } else {
+            val hostname = args[0]
+            val endpoint = "https://$hostname.herokuapp.com/api/v1/"
+            buildNightscoutResponse(endpoint, event)
```

Comment:
Move `buildNightscoutResponse` to outside if-else statement and `endpoint` declaration before it.

Reply:
Will now catch if a mentioned user doesn't have a host set and throw IllegalArgumentException.